tests/test_protocol.py

```diff
@@ -0,0 +1,63 @@
+"""
+Runs tests that ensure protocol invariants
+
+TODO add other tests including:
+- some CI on postSignatures
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+
+import avro.schema
+
+import compile_schemas
+
+
+class TestValidateSchemas(unittest.TestCase):
+    """
+    Ensure the schemas conform to certain rules
```

Comment:
Any rules in particular? It looks like the only rule here is the nulls in unions thing. When we get more rules, do we want one big "follows the rules" test, or do we want a test for each rule?